### PR TITLE
Add commandline options to ini file

### PR DESCRIPTION
**Pull request by thokkat**
Commandline options are now accessible from the ini file as proposed here https://github.com/Try/OpenGothic/pull/84. In case there are conflicting options set command line takes precedence over ini settings.

I renamed the `Gothic.ini` in the executable path to `OpenGothic.ini` to avoid confusion and to emphasize that those settings are only used in OpenGothic. The ini is now generated at game start by default if it doesn't exist already.

### CODE REVIEW

**Comment by Try** [comment]
> I renamed the Gothic.ini in the executable path to OpenGothic.ini 

Can't say that I'm liking this decision. Originally idea for *.ini file in OpenGothic is to organize them in chains:
* [0] - default settings file (virtual)
* [1] - Gothic.ini from installation folder
* [2] - Gothic.ini from OpenGothic folder

**Review by Try — COMMENTED** [review]
After looking into this PR, it doesn't feel like valuable change: almost all command-line arguments are set for debugging purpose and putting it to 8.ini only complicates things.

**Comment by thokkat** [comment]
I was just referring to https://github.com/Try/OpenGothic/pull/84#issuecomment-672120390 and didn't think to much about the usefulness for debug options. Does it make sense for non debug stuff or is there not enough value for like 3 settings.

Your original proposal was to add an `OpenGothic` section in the `gothic.ini` in the installation folder. But when you change e.g. sound volume this setting is written into the `gothic.ini` in the OpenGothic folder. Thought it's better to have all OpenGothic related options there.

File: game/gothic.cpp [hunk 1]
```diff
@@ -608,6 +653,8 @@ int Gothic::settingsGetI(std::string_view sec, std::string_view name) {
   }
 
 void Gothic::settingsSetI(std::string_view sec, std::string_view name, int val) {
+  if (!instance->defaults->has(sec,name))
```
[the remark this answers is not in the document]
Comment (thokkat) [reply game/gothic.cpp:656]: To avoid having unused values written to ini.

**Comment by thokkat** [comment]
@Try Should i scratch everything or only debug stuff and move non debug options to gothic.ini in a `OPEN_GOTHIC` section?

Regarding `SytemPack.ini` for the `HideFocus` option, is  this supposed to work like the `gothic.ini`? Add 
``SystemPackIniFile.reset(new IniFile(nestedPath({u"system",u"SytemPack.ini"},Dir::FT_File)));``and query it via the  `settingsGet` routines?

**Comment by Try** [comment]
> Should i scratch everything or only debug stuff and move non debug options 

How much is there to remain, if debug stuff is removed?
```
-g
-nofrate (aka HideFocus)
-window (aka zStartupWindowed)
-rt (maybe?)
```

> ..  supposed to work like the gothic.ini? .. and query it via the settingsGet routines?

Depends on how it works in SystemPack.ini. I don't know for sure, but probably it's not path of standard settings routing.
`settingsGet` practically mean that script will have access to those values - probably not correct behavior.
So ,likely it's:
```
if( "-window" is not specifyed )
  systemPackIniFile->getI(...);
```

**Comment by thokkat** [comment]
> ```
> -g
> -nofrate (aka HideFocus)
> -window (aka zStartupWindowed)
> -rt (maybe?)
> ```

`dx12` maybe this is better than vulkan on some hardware? But since `HideFocus, windowed` and `nofrate` already have entries only `-g, -rt` and `dx12` would remain. Probably gets more useful once there're more graphics options to play with.

**Comment by Try** [comment]
I think for graphics we should new sections/value in *.ini file.
something like that:
```
[ENGINE]
zDetailTexturesEnabled=1 ; existing settings
tDepthOfFieldEnabled = 1 ; new stuff
```

**Pull request closed without merging**